src/App.tsx

```diff
@@ -17,18 +60,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onSelectChange={setSelectedFilter}
+                onInputChange={setInputFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={todos}
+                  onSelectedTodo={setSelectedTodo}
+                  onModalOpened={setModalOpened}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          onSelectedTodo={setSelectedTodo}
+          onModalOpened={setModalOpened}
```

Comment:
Consider following the naming conventions for event handler functions. For example, use 'handleModalOpened' instead of 'onModalOpened'.